### ISSUE TITLE
Sword and Arrows not working

### ISSUE THREAD

**Issue by grendelbiter** [thestonefox/VRTK#665]
Hello,
I apologize if this is the wrong place to post this. I have a problem that is driving me crazy. I have just started last week with game development. I have zero experience in coding, I'm more of a graphics guy. I'm very grateful for this collection of scripts that enables me to do stuff. That said, I made a scene with a custom bow and arrow and a sword and a couple of cubes. It worked fine. I started adding all kinds of plugins and stuff from the asset store. At some point I tried out the sword and it didn't work...same with the bow and arrow. I have no idea when it stopped working, I didn't touch them for a while. Whenever I try to hit a Breakable Cube with the sword it displays this message:

NullReferenceException: Object reference not set to an instance of an object
VRTK.Examples.Breakable_Cube.GetCollisionForce (UnityEngine.Collision collision) (at Assets/VRTK/Examples/Resources/Scripts/Breakable_Cube.cs:26)
VRTK.Examples.Breakable_Cube.OnCollisionEnter (UnityEngine.Collision collision) (at Assets/VRTK/Examples/Resources/Scripts/Breakable_Cube.cs:16)

 I checked those lines in the code (16, 26) and they are exactly the same as on the asset store. I deleted and redownloaded VRTK...didn't help. When I recreate the scene in another project it works fine (at least the sword, the bow is a PITA to build)

The only thing I can think of that may cause it is some plugin I downloaded. There's one I'm not even able to delete. So do I have to make a new project or can this one be rescued?

**Comment by JoeHart** [comment]
Without seeing the code or knowing exactly what plugins you added it's pretty hard to narrow down.

Best thing to do would be to slowly remove the things you have added, possibly simply deactivating them rather then fully deleting them until they work again. This trial and error approach should allow you to figure out whats interfering with what.

Note: I'm not a contributor to VRTK, just a fellow Unity VR dev.

**Comment by grendelbiter** [comment]
No plugins that have anything to do with the controllers or anything VR. Plugins I added:

Toon Forest free set
A HDR skybox
Cafofo game music
Aquas water shader - this one also automatically downloaded standard assets. Unity complained quite a lot when doing that but no red errors.

Cloudstoy - I can't delete this one, that's driving me nuts too. It just automatically redownloads itself. I've removed everything from it from the scene.

**Comment by JoeHart** [comment]
Looking at the lines of the code mentioned in the error it looks like they reference the name of the sword to check for collision.

`            if ((collision.collider.name.Contains("Sword") && collision.collider.GetComponent<Sword>().CollisionForce() > breakForce))
`

Did you change the name of the Sword?

**Comment by grendelbiter** [comment]
Didn't touch it at all. I did add an axe though with the sword script, named axe. Come to think of it that's when I noticed the sword wasn't working cause the axe wasn't so I tried the sword. I removed the script from the axe though and it's still the same. Doesn't explain why arrows wouldn't work either. One thing I should mention is I get the haptic feedback and the box does disappear, but in a wierd way, like it just slides under the terrain everytime in the same direction no matter which way you hit it.

**Comment by thestonefox** [comment]
Depending on what version you're on, you may not have the namespaces in the example files. therefore if you have another script called `Sword.cs` then it will clash and break the example.

**Comment by grendelbiter** [comment]
I have no idea what that means but I guess just download the newest version here?

**Comment by thestonefox** [comment]
Do you have two scripts named `Sword.cs` ?

**Comment by grendelbiter** [comment]
No I used the one from the VRTK folder, I didn't copy any scripts.

**Comment by thestonefox** [comment]
Just try downloading the latest version then and see how you get on.

**Comment by grendelbiter** [comment]
The problem still persists. I've managed to delete all downloaded assets except Steam VR and VRTK. Even CloudToy. Why I was able to just delete it this time I don't know, maybe because I deleted it last. I made a fork of this depository (just figured that out) and copied the VRTK directory from the zip to my project/assets folder. This time there is no error message though. At least in one scene. In another test scene I made there is a Nullexception, same message every time I hit a box. No message with arrows. The boxes just get knocked through the floor and fall, I watched it from a ledge.

**Comment by thestonefox** [comment]
I'd recommend downloading VRTK from github and opening it as a fresh project in unity then importing the SteamVR asset from the Asset Store and see how you get on.

**Comment by grendelbiter** [comment]
Yeah that'll fix it for sure, I already started a new project with the sword set up and it works fine. It'll be a real pain in the ass to set up the bow again though. I do have all the settings now so it'll be a bit faster but still. I tried to import the bow asset but it doesn't import the model, maybe I have to import that first. Still wondering what caused that behavior. Thanks for the troubleshooting.
